core/src/main/java/com/kin/ecosystem/core/accountmanager/AccountManagerImpl.java

```diff
@@ -96,13 +94,18 @@ public void retry() {
 		}
 	}
 
+	@Override
+	public void logout() {
+		accountState.removeAllObservers();
```

Comment:
no need to removeAllObservers

Reply:
why not, it will call add again so we don't need double events.